Doist/typist
# fix(deps)!: update unified ecosystem dependencies

## Overview

Unified.js v11 has been released, and as you might know, unified is "a collective of 500+ free and open source packages that work with content as structured data (ASTs)".

This PR will take care of bumping all related peer dependencies to their latest major versions in one fell swoop. This is required because the latest versions have some breaking changes that affect us, and while some dependencies don't seem to cause any issues, it's best if we only use versions that were meant to work with each other.

Here's a list of all dependencies that we are using, and which ones have already been upgraded:

```[tasklist]
### Dependencies
- [x] `mdast-util-gfm-autolink-literal`
- [x] `mdast-util-gfm-strikethrough`
- [x] `micromark-extension-gfm-autolink-literal`
- [x] `micromark-extension-gfm-strikethrough`
- [x] `react-markdown`
- [x] `rehype-minify-whitespace`
- [x] `rehype-raw`
- [x] `rehype-stringify`
- [x] `rehype`
- [x] `remark-breaks`
- [x] `remark-gfm`
- [x] `remark-rehype`
- [x] `remark`
- [x] `unified`
- [x] `unist-util-is`
- [x] `unist-util-remove`
- [x] `unist-util-visit`
```

## PR Checklist


## Test plan

- Open the preview Storybook deployed to Netlify
- Open the `Rich-text → Functions → Commands` story
- Click the `insertMarkdownContent` button
    - [x] Observe that the editor content is rendered as expected
- Use the `content` control input, and type whatever **Markdown** you want
    - [x] Observe that the editor content is always rendered as expected

> [!NOTE]
> We are only using Unified in the HTML serializer (for now), so changes in this PR do not affect the "Markdown Output" window, only the "Typist Editor" window (that is, Markdown content that needs to be parsed into HTML to be fed into Tiptap).
